### DISCUSSION TITLE
WEBP target-size has no effect?

### DISCUSSION THREAD

**Discussion by aybli** [ImageMagick/ImageMagick#4923] [Help]
I'm using the latest version on Win10 but also tried with an older version.

All of these commands produce the same image:

```
magick convert image.jpg -define webp:target-size=50000 image1.webp
magick convert image.jpg image2.webp
magick convert image.jpg -define webp:target-size=100 image3.webp
```

It's the same for "target-psnr=value".

No problem with `cwebp.exe -size 50000 image.jpg -o image.webp`

**Comment by fmw42** [comment]
On Imagemagick 7, use magick not magick convert.  Does that help?  If not, are you using the latest webp delegate library?

**Reply by aybli** [reply]
> On Imagemagick 7, use magick not magick convert. Does that help? If not, are you using the latest webp delegate library?

Same results when using without convert.
Not sure how to check this. I have the latest `libwebp-1.2.2-windows-x64` downloaded and added to my PATH.

**Reply by fmw42** [reply]
The IM developers will need to help you further. I do not use Windows.

**Reply by aybli** [reply]
I'm using ImageMagick-7.1.0-27-Q16-HDRI-x64-dll

I guess it doesn't have anything to do with that... here's the output of "magick -list format"
`WEBP* WEBP      rw+   WebP Image Format (libwebp 1.0.3 [020E])`

This version of libwebp is from 2019 but it also supports the -size (webp:target-size=value) and -psnr (webp:target-psnr=value) options.

**Reply by fmw42** [reply]
I do not know the minimum webp version that Imagemagick needs.  But why don't you try to upgrade it, there is a more current version, 1.2.2.  See https://developers.google.com/speed/webp/docs/api and https://developers.google.com/speed/webp/download

**Comment by rawriddims** [comment]
Hello, I did post a duplicate of this issue before, sorry.
Anyway, I'm having the same problem: when I'm running `magick in.jpg -resize 1768 -define webp:target-size=235000 out.webp` webp:target-size= has no effect.

On the other hand when I convert to webp with -quality is properly working.

I do have ImageMagick 7.1.0-36 Q16-HDRI x64 be0bdc3:20220530
with libwebp 1.2.2 [020F]
on Windows10 x64

**Comment by m3at** [comment]
I can confirm having the same issue in MacOS 12.3 (latest imagemagick: 7.1.0-33)

**Comment by rawriddims** [comment]
At this point I wonder if there is anyone who has managed to make it work. It would be nice to hear what an IM developer thinks - if he/she reads us

**Comment by dlemstra** [comment]
We have read this but haven't had/made time to investigate this.

**Reply by rawriddims** [reply]
Thanks

**Reply by dlemstra** [reply]
The cause of this has been found. The command line sets `webp:pass` to 6 and this will be added in the next release. A temporary workaround would be to set `webp:pass` for now.

**Reply by rawriddims** [reply]
Like this? `magick in.jpg -define webp:pass=6 -define webp:target-size=200000 out.webp`
It works, and as stated "The compressor makes several passes of partial encoding in order to get as close as possible to this target" - even if the output image is 350kb around instead of 200

**Reply by dlemstra** [reply]
That would be the workaround for now. In the next release you don't need to add `-define webp:pass=6` because this will be added automatically.

**Reply by rawriddims** [reply]
Splendid! Thank you very much!

**Reply by contentfree** [reply]
Adding `-define webp:pass=6` doesn't make a difference for me (macOS; IM 7.1.0-53; libwebp 1.2.4). The output webp is the same size whether or not I give webp:target-size and/or webp:pass (including neither…)

**Reply by contentfree** [reply]
Interestingly, including webp:pass and webp:target-size makes it work in IM 6.9.

**Comment by Xyn0gen** [comment]
I noticed that target-size would not work with jpg images that have a quality of 100 when checking with `magick identify -verbose image.jpg`. Adding `-define webp:lossless=false` would allow target-size to work as expected. Using `ImageMagick 7.1.0-61 Q16-HDRI x64 d396287:20230205` on Windows.

**Reply by GrauBlitz** [reply]
This seems to be the solution for me, looks like IM ignores a target size if lossy encoding is not explicitly allowed with `-define webp:lossless=false`, thanks!.

ImageMagick 7.1.2-17 on Manjaro/Arch/Linux
